hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistLazyInitializer.java

```diff
@@ -57,6 +57,11 @@ public Object invoke(
 			final Method proceed,
 			final Object[] args) throws Throwable {
 		if ( this.constructed ) {
```

Comment:
@gbadner, just try your suggestion and invoke `thisMethod` and see what happens!

Now more seriously,  although both methods have the same name (that is expected) one refers to the actual class and the other to the proxy class. If you invoke the method on the class you get routed back to the initializer and end up in an infinite recursion.

I don't get your first comment though. A proxy gets initialized as soon as one of it's methods is called. What I'm doing here is disable that for the methods added by bytecode enhancement, otherwise we get the behavior described in the issue (a load triggered when it's not supposed).